samples/detect.v2beta1.js

```diff
@@ -416,26 +360,20 @@ function detectIntentwithTexttoSpeechResponse(
     },
   };
 
-  sessionClient
-    .detectIntent(request)
-    .then(responses => {
-      console.log('Detected intent:');
-      const audioFile = responses[0].outputAudio;
-      fs.writeFile(outputFile, audioFile, 'binary', err => {
-        if (err) {
-          console.error('ERROR:', err);
-          return;
-        }
-        console.log(`Audio content written to file: ${outputFile}`);
-      });
-    })
-    .catch(err => {
+  const responses = await sessionClient.detectIntent(request);
+  console.log('Detected intent:');
+  const audioFile = responses[0].outputAudio;
+  await util.promisify(fs.writeFile)(outputFile, audioFile, 'binary', err => {
```

Comment:
If you use promisify, you don't need the callback.  
https://nodejs.org/api/util.html#util_util_promisify_original